cmd/registry/patterns/resources.go

```diff
@@ -321,6 +331,10 @@ func (ar ArtifactName) ParentName() ResourceName {
 		return SpecName{
 			Name: spec,
 		}
+	} else if rev, err := names.ParseSpecRevision(parent); err == nil {
+		return SpecName{
+			Name: rev.Spec(),
+		}
```

Comment:
Shouldn't this be:

```
return SpecName{
			Name: rev.Spec(),
                         RevisionID: rev.RevisionID,
		}

```


Should we also support revision collections in the previous block for matching collections?
Example: `projects/demo/locations/global/apis/-/versions/-/specs/-@-/artifacts/-` 
Not sure if the names package supports this yet.

Reply:
Looking into this... Adding RevisionID to this breaks [this controller test](https://github.com/apigee/registry/blob/1777bc258f5719f80dd4cb9d3b8dd4fa285c78de/cmd/registry/controller/controller_test.go#L659) - it generates double the actions: a set of update actions for SpecRevisions and a set of create actions for Specs. So, I'm working to understand that.